traffic_ops/traffic_ops_golang/apitenant/tenant.go

```diff
@@ -143,10 +155,22 @@ func (ten *TOTenant) Read() ([]interface{}, error, error, int) {
 			// root tenant has no parent
 			continue
 		}
+		found = true
 		p := *tenantNames[*t.ParentID]
+		// In case of a bulk read, even if one of the items has a "lastUpdated" time that is after whats supplied in the request,
+		// we send back the entire array of results
+		if !t.LastUpdated.Before(modifiedSince) {
+			modified = true
+		}
 		t.ParentName = &p // copy
 	}
-	return tenants, nil, nil, http.StatusOK
+	// If the modified flag stayed false throughout (meaning that all the items' "lastUpdated" time is before whats supplied in the request),
+	// we send back a 304, with an empty response
+	if modified == false && found == true {
+		code = http.StatusNotModified
+		tenants = []interface{}{}
+	}
+	return tenants, nil, nil, code
```

Comment:
this logic looks like if no IMS header is sent in the request it will send back a Not Modified response.